the-collab-lab/tcl-26-smart-shopping-list
# Delete shopping item

## Description



We've added the ability to delete a Shopping Item by adding a "Delete" button next to each item. A modal appears if you attempt to delete an item asking you to confirm that you'd like to delete it. If you select to delete the item, it removes the item from Firestore, and there is a handleCloseModal to close the modal.

We've also tried to make this modal accessible by allowing you to navigate to it and use it with your keyboard. We've used the following as resources for developing accessible modals:
- Aria practices: https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html
- Keyboard navigation: https://tinloof.com/blog/how-to-create-an-accessible-react-modal/

## Related Issue



Closes #11 : As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.

## Acceptance Criteria



- User is able to delete an item from the shopping list
- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
- Deletion should cause the associated record(s) in the database to be deleted


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before



![image](https://user-images.githubusercontent.com/61371242/129829741-a1792eb9-0527-4338-a8fe-a5790fc7db44.png)


### After



![image](https://user-images.githubusercontent.com/61371242/129829761-0571b4fb-f8e1-49db-9ccb-b663356ac5fb.png)


## Testing Steps / QA Criteria



Functionality:
1. Click "Delete" next to an item.
2. In the modal, clicking "No, Cancel" will close the modal.
3. In the modal, clicking "Yes, Delete" will remove the item from the Shopping List.
4. The item is no longer in Firestore.

Accessibility steps:
1. You should be able to use your keyboard to navigate and select "Delete".
2. When the modal opens, the focus will be on "No, Cancel".
3. You can tab in the modal which will navigate you to either "No, Cancel" or "Yes, Delete". 
4. You can use Enter to select your option.
5. You can escape to close the modal.
